src/components/OSCALComponentDefinition.js

```diff
@@ -17,12 +18,15 @@ const useStyles = makeStyles((theme) => ({
 export default function OSCALComponentDefinition(props) {
   const [error, setError] = useState(null);
   const [isLoaded, setIsLoaded] = useState(false);
+  const [inheritedProfilesAndCatalogs, setInheritedProfilesAndCatalogs] =
+    useState({});
   const classes = useStyles();
 
   useEffect(() => {
     OSCALComponentResolveSources(
       props.componentDefinition,
       props.parentUrl,
+      setInheritedProfilesAndCatalogs,
```

Comment:
It seems a bit odd to send in essentially another `onSuccess` function here.  Should we consider simply adding some form of `inheritedProfilesAndCatalogs` argument to the existing `onSuccess` function and call `setInheritedProfilesAndCatalogs` within that?